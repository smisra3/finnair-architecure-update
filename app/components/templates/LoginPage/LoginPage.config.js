
import initialActions from "./LoginPage.action";
import reducer from "./LoginPage.reducer";
import saga from "./LoginPage.saga";
import { mapStateToProps, mapDispatchToProps } from './LoginPage';

export default {
  key: "login",
  reducer,
  saga,
  initialActions,
  mapStateToProps,
  mapDispatchToProps
};
