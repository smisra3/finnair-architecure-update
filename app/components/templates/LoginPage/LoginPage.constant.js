// TODO: Remove exception
/* eslint-disable import/prefer-default-export */
export const LOAD_SIGNUP_SLOT = 'FIL/LOGINPAGE/LOAD_SIGNUP_SLOT';
export const LOAD_SIGNUP_SLOT_SUCCESS = 'FIL/LOGINPAGE/LOAD_SIGNUP_SLOT_SUCCESS';
export const FAILURE = 'FIL/LOGINPAGE/FAILURE';
