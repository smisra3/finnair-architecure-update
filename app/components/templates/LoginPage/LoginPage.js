// @flow
import { PureComponent } from "react";
import { connect } from "react-redux";
import Router from "next/router";
import styled from "styled-components";
import get from "lodash/get";
import Layout from "../Layout";
import enhance from "../../../lib/dynamicStore";
import SignInContainer from "../../organisms/SignIn";
import reducer from "./LoginPage.reducer";
import saga from "./LoginPage.saga";
import initialActions from "./LoginPage.action";
import styles from "./LoginPage.style";
import { MY_ACCOUNT } from "../../../routes";
import HeadTag from "../../atoms/HeadTag";
import { loginPageMetaLabels, metaInfo } from "./LoginPage.static.data";

class Login extends PureComponent<any, any> {
  static getInitialProps = ({ res, isServer, store }: any) => {
    const isUserLoggedIn = get(store.getState(), [
      "global",
      "globalData",
      "sessionInfo",
      "isUserLoggedIn"
    ]);
    if (isUserLoggedIn) {
      if (isServer) {
        res.redirect(MY_ACCOUNT);
      }
    }
  };
  static getProps(props) {
    return props;
  }
  sendToMyAccount = defaultRoute => {
    Router.replace({ pathname: MY_ACCOUNT, query: { page: defaultRoute } }, `${MY_ACCOUNT}`).then(
      () => {
        window.scrollTo(0, 0);
      }
    );
  };

  componentDidMount() {
    console.log("hello", this.props);
  }

  componentDidUpdate(prevProps, prevState, snapshot) {
    const { isUserLoggedIn } = this.props;
    if (prevProps.isUserLoggedIn !== isUserLoggedIn) {
      if (isUserLoggedIn) {
        this.sendToMyAccount(MY_ACCOUNT);
      }
    }
  }

  render() {
    const { className } = this.props;
    Login.getProps(this.props);
    return (
      <Layout title="SIGN IN">
        <div className={className}>
          <HeadTag
            description={loginPageMetaLabels.loginPageTitleText}
            image="image"
            title={loginPageMetaLabels.loginPageTitleText}
            url="canonical url"
            siteName={metaInfo.SITE_NAME}
            card="card"
          />
          <h1 className="visually-hidden">{loginPageMetaLabels.loginPageTitleText}</h1>
          <section className="login-container" />
        </div>
      </Layout>
    );
  }
}

/* istanbul ignore next */
const mapStateToProps = (state: Map): MapStateToProps => ({
  data: get(state, ["global"]),
  deviceType: get(state, ["global", "globalData", "deviceType"]),
  activeBrand: get(state, ["global", "globalData", "activeBrand"]),
  isUserLoggedIn: get(state, ["global", "globalData", "sessionInfo", "isUserLoggedIn"])
});

/* istanbul ignore next */
const mapDispatchToProps = (dispatch): void => ({});

const LoginStyled = styled(Login)`
  ${styles};
`;
//const isPropsRendered = Login.getProps() === undefined;

// export default connect(state => state)(Login);
export default enhance(LoginStyled, {
  key: "login",
  reducer,
  saga,
  initialActions,
  mapStateToProps,
  mapDispatchToProps
});

export { Login as LoginVanilla };
