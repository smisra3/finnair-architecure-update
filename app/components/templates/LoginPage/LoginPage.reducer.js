// @flow
import { combineReducers } from 'redux';

export const loginPageData = (state = {}, action) => {
  return state;
};

export default combineReducers({
  loginPageData,
});
