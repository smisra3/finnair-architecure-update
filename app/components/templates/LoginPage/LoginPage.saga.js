/**
 * Sign In template Saga Watcher
 */
export default function* SignInTemplateSaga() {
 
}
