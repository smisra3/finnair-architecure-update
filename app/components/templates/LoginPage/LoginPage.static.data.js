import { loginPageLabels } from "../../../locales/en-US";

const loginPageMetaLabels = {
  loginPageTitleText: "FINNAIR" //loginPageLabels.loginPageTitleText
};

const metaInfo = {
  SITE_NAME: "FINNAIR"
};

export { loginPageMetaLabels, metaInfo };
