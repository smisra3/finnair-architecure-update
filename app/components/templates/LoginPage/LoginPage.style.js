import { css } from 'styled-components';

const commonStyles = css`
  .login-container {
    margin: 60px auto;
    max-width: 480px;
    @media (max-width: 767px){
      margin: 20px auto;
      max-width: 90%;
    }
  }
`;

export default commonStyles;
