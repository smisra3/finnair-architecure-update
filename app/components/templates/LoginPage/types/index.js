// @flow
export type Props = {
  deviceType: string,
};

export type MapStateToProps = {
  deviceType: string,
};

