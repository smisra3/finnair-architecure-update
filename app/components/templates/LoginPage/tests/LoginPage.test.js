import { shallow } from 'enzyme';
import { LoginVanilla } from '../LoginPage';
import { MOBILE, DESKTOP } from '../../../../constants';
import ForgotPassword from '../../../molecules/ForgotPassword';
import SignIn from '../../../organisms/SignIn';

const props = {
  deviceType: MOBILE,
  forgotReset: jest.fn(),
  forgotSubmit: jest.fn(),
};

describe('SignIn Component', () => {
  let signIn = '';
  test('should render correctly', () => {
    signIn = shallow(<LoginVanilla {...props} />);
    expect(signIn).toMatchSnapshot();
  });

  test('should render correctly', () => {
    props.deviceType = DESKTOP;
    signIn = shallow(<LoginVanilla {...props} />);
    expect(signIn).toMatchSnapshot();
  });

  test('Should trigger showForgetModal', () => {
    const shallowComponent = shallow(<LoginVanilla {...props} />);
    const spy = jest.spyOn(shallowComponent.instance(), 'showForgetModal');
    shallowComponent.setState({ visibleforgetModal: true });
    shallowComponent.find(SignIn).prop('showForgetModal')();
    expect(spy).toHaveBeenCalled();
  });
  test('Should trigger forgotSubmit', () => {
    const shallowComponent = shallow(<LoginVanilla {...props} />);
    const spy = jest.spyOn(shallowComponent.instance(), 'forgotSubmit');
    shallowComponent.setState({ visibleforgetModal: true });
    shallowComponent.find(ForgotPassword).prop('forgotSubmit')('test@test.com');
    expect(spy).toHaveBeenCalled();
  });
  test('Should trigger hideForgotModal', () => {
    const shallowComponent = shallow(<LoginVanilla {...props} />);
    const spy = jest.spyOn(shallowComponent.instance(), 'hideForgotModal');
    shallowComponent.setState({ visibleforgetModal: true });
    shallowComponent.find(ForgotPassword).prop('closeModal')();
    expect(spy).toHaveBeenCalled();
  });
});
