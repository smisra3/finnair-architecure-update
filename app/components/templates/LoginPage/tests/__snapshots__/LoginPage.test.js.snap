// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn Component should render correctly 1`] = `
<Connect(Layout)
  className="row"
  title="SIGN IN"
>
  <styled.div
    className="col-xs-12 row"
  >
    <HeadTag
      card="card"
      description="Login | Ann Taylor Factory"
      image="image"
      siteName="factory.anntaylor.com"
      title="Login | Ann Taylor Factory"
      url="canonical url"
    />
    <h1
      className="visually-hidden"
    >
      Login | Ann Taylor Factory
    </h1>
    <ContentSlotUnsafeHTML
      className="col-xs-12"
      hasScript={true}
      wrapper={null}
    />
    <Connect(Styled(SignUp))
      className="col-sm-offset-2 col-sm-4 col-xs-12"
    />
    <Connect(Styled(SignIn))
      className="col-sm-offset-1 col-sm-4 col-xs-12"
      showForgetModal={[Function]}
    />
  </styled.div>
</Connect(Layout)>
`;

exports[`SignIn Component should render correctly 2`] = `
<Connect(Layout)
  className="row"
  title="SIGN IN"
>
  <styled.div
    className="col-xs-12 row"
  >
    <HeadTag
      card="card"
      description="Login | Ann Taylor Factory"
      image="image"
      siteName="factory.anntaylor.com"
      title="Login | Ann Taylor Factory"
      url="canonical url"
    />
    <h1
      className="visually-hidden"
    >
      Login | Ann Taylor Factory
    </h1>
    <ContentSlotUnsafeHTML
      className="col-xs-12"
      hasScript={true}
      wrapper={null}
    />
    <Connect(Styled(SignIn))
      className="col-sm-offset-1 col-sm-4 col-xs-12"
      showForgetModal={[Function]}
    />
    <Connect(Styled(SignUp))
      className="col-sm-offset-2 col-sm-4 col-xs-12"
    />
  </styled.div>
</Connect(Layout)>
`;
